src/store/app.js

```diff
@@ -30,34 +30,70 @@ async function findEntry(email, password) {
 async function authenticate(entry) {
   if (!entry || !entry.dn) return
   const res = await axios.post(`/hdap/${entry.dn}?_action=authenticate`, {
-    password: `${entry.password}`
-  }, {
-    headers: { 'Content-Type': 'application/json' }
-  })
-  return res.data.access_token
+      password: `${entry.password}`
+    }, {
+      headers: { 'Content-Type': 'application/json' }
+    })
+  return res.data
+}
+
+function getExpInSecsSinceEpoch(secondsFromNow) {
+  // Make sure numbers are numbers here. Good grief!
```

Comment:
At first, I tried to do everything with JavaScript Date objects. But they were getting stored as `null`, so I converted everything to seconds since the epoch. Maybe I should've made everything date strings?

Reply:
Going to take another look at this later when moving to a separate stateful HDAP API in JavaScript